doc/source/whatsnew/v1.2.0.rst

```diff
@@ -316,6 +316,7 @@ Performance improvements
 - Performance improvement in :meth:`RollingGroupby.count` (:issue:`35625`)
 - Small performance decrease to :meth:`Rolling.min` and :meth:`Rolling.max` for fixed windows (:issue:`36567`)
 - Performance improvement in :class:`ExpandingGroupby` (:issue:`37064`)
+- Performance improvement in :meth:`pd.DataFrame.groupby` for ``float`` ``dtype`` (:issue:`28303`)
```

Comment:
can you also add that float based indexes sort ordering for ties could change versus prior results (e.g. like the tests you changed for a float based .value_counts() result), basically the hash changed (on purpose).

Reply:
Fixed here: https://github.com/pandas-dev/pandas/pull/36729/commits/f2496381d1dd088df60f29e6a9604dce30abaf1f#diff-9f46290979b546104c6627ab86918577bbfc89deb3f896fc1bf53d67909baad1R379